subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -2087,30 +2094,35 @@ static int parse_write_op(struct lwm2m_message *msg, uint16_t format)
 		 * number.
 		 */
 		block_ctx->expected += GET_BLOCK_SIZE(block_opt) - block_ctx->ctx.block_size + 1;
-
-		/* Handle blockwise 1 (Part 1): Set response code */
-		if (!last_block) {
-			msg->code = COAP_RESPONSE_CODE_CONTINUE;
-		}
 	}
 
 	r = do_write_op(msg, format);
 
 	/* Handle blockwise 1 (Part 2): Append BLOCK1 option / free context */
 	if (block_ctx) {
-		if (r >= 0 && !last_block) {
-			/* More to come, ack with correspond block # */
+		if (r >= 0) {
+			/* Add block1 option to response.
+			 * As RFC7959 Section-2.3, More flag is off, because we have already
+			 * written the data.
+			 */
 			r = coap_append_block1_option(msg->out.out_cpkt, &block_ctx->ctx);
 			if (r < 0) {
 				/* report as internal server error */
-				LOG_ERR("Fail adding block1 option: %d", r);
+				LOG_DBG("Fail adding block1 option: %d", r);
```

Comment:
Why change the log level?

Reply:
I thought that this is so rare, almost impossible to fail, that I don't want to accumulate these error messages into flash.
CoAP packets have statically allocated sizes, these options will always fit. If they don't it should be visible from outside the system, in which case someone will take a look with debug logs enabled.